dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/doi/DOIBackgroundReservationServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package edu.harvard.iq.dataverse.doi;
+
+import com.google.common.collect.Lists;
+import edu.harvard.iq.dataverse.globalid.DOIDataCiteServiceBean;
+import edu.harvard.iq.dataverse.persistence.GlobalId;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetRepository;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class DOIBackgroundReservationServiceTest {
+
+    @Mock
+    private SettingsServiceBean settingsServiceBean;
+
+    @Spy
+    private Timer timer;
```

Comment:
Why we use spy instead of mock? If spy is necessary then probably we should assure that no thread created by Timer object is running between tests